Incompatible `Filter.Invert` data type when using rebuy-de/aws-nuke config

While migrating from rebuy-de/aws-nuke to libnuke in https://github.com/fluxcd/test-infra/pull/39, I encountered a panic due to the invert filter at https://github.com/ekristen/libnuke/blob/v0.15.0/pkg/filter/filter.go#L210 for a boolean value. Changing the invert value from boolean to string resolves the issue.

```console
$ aws-nuke run -c nuke-cfg.yaml 
panic: interface conversion: interface {} is bool, not string [recovered]
	panic: interface conversion: interface {} is bool, not string [recovered]
	panic: interface conversion: interface {} is bool, not string

goroutine 1 [running]:
main.main.func1()
	/github.com/ekristen/aws-nuke/main.go:26 +0x54
panic({0x541b5e0?, 0xc00044b770?})
	/go/src/runtime/panic.go:770 +0x132
gopkg.in/yaml%2ev2.handleErr(0xc00059f118)
	/go/pkg/mod/gopkg.in/yaml.v2@v2.4.0/yaml.go:249 +0x6d
panic({0x541b5e0?, 0xc00044b770?})
	/go/src/runtime/panic.go:770 +0x132
github.com/ekristen/libnuke/pkg/filter.(*Filter).UnmarshalYAML(0xc0001144e0, 0xc000075d80)
	/go/pkg/mod/github.com/ekristen/libnuke@v0.15.0/pkg/filter/filter.go:210 +0x4d0
gopkg.in/yaml%2ev2.(*decoder).callUnmarshaler(0xc0001143c0, 0xc0005496c0, {0x7b17601f0198, 0xc0001144e0})
	/go/pkg/mod/gopkg.in/yaml.v2@v2.4.0/decode.go:270 +0x94
...
```

It may have been an intentional incompatibility. But since I couldn't find any mention of it, thought of reporting it.
Please close the issue if this is known and intentional.

Thanks for a nice alternative tool and library :slightly_smiling_face:

--- a/pkg/filter/filter.go
+++ b/pkg/filter/filter.go
@@ -207,7 +207,12 @@ func (f *Filter) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if m["invert"] == nil {
 		f.Invert = ""
 	} else {
-		f.Invert = m["invert"].(string)
+		switch v := m["invert"].(type) {
+		case bool:
+			f.Invert = strconv.FormatBool(v)
+		default:
+			f.Invert = v.(string)
+		}
 	}
 
 	return nil

--- a/pkg/filter/filter_test.go
+++ b/pkg/filter/filter_test.go
@@ -218,8 +218,14 @@ func TestFilter_UnmarshalFilter(t *testing.T) {
 			error: true,
 		},
 		{
-			name:     "invert",
+			name:     "invert-truthy",
 			yaml:     `{"type":"exact","value":"foo","invert":"true"}`,
+			match:    []string{"foo"},
+			mismatch: []string{"bar", "baz"},
+		},
+		{
+			name:     "invert-true",
+			yaml:     `{"type":"exact","value":"foo","invert":true}`,
 			match:    []string{"foo"},
 			mismatch: []string{"bar", "baz"},
 		},

